packages/netsuite-adapter/src/change_validators/remove_list_item_without_scriptid.ts

```diff
@@ -0,0 +1,170 @@
+
+const getRoleListPath: GetListPath = () => [
+  PERMISSIONS,
+  PERMISSION,
+]
+
+const getRolePermissionList:GetItemList = (
+  instance: InstanceElement,
+): RolePermissionObject[] => {
+  const listPathValue = _.get(instance.value, getRoleListPath())
+  if (_.isPlainObject(listPathValue)) {
+    return Object.values(listPathValue)
+      .filter(isRolePermissionObject)
+  }
+  return [] as RolePermissionObject[]
+}
+
+const getRolePermkey: GetItemString = (
+  permission: RolePermissionObject
+): string => {
+  const permkey = permission[PERMKEY]
+  if (_.isString(permkey)) {
+    return permkey
+  }
+  return permkey.value
+}
+
+const getRolePermissionByName = (
+  role: InstanceElement,
+  id: string,
+): RolePermissionObject | undefined => (
+  _.isPlainObject(role.value[PERMISSIONS]?.[PERMISSION])
+    ? Object.values(role.value[PERMISSIONS]?.[PERMISSION])
+      .filter(isRolePermissionObject)
+      .find(permObj => getRolePermkey(permObj) === id)
+    : undefined
+)
+
+const getRoleMessage = (removedListItems: string[]): string =>
+  getMessageByElementNameAndListItems(PERMISSION, removedListItems)
+
+export const roleGetters: ItemListGetters = {
+  getItemList: getRolePermissionList,
+  getItemString: getRolePermkey,
+  getItemByID: getRolePermissionByName,
+  getListPath: getRoleListPath,
+  getDetailedMessage: getRoleMessage,
+}
+
+export const getGettersByType = (
+  typename: string,
+): ItemListGetters | undefined => {
+  if (typename === ROLE) {
+    return roleGetters
+  }
+  return undefined
+}
+const getIdentifierList = (
+  instance: InstanceElement,
+  getters: ItemListGetters,
+): string[] =>
+  getters.getItemList(instance).map(getters.getItemString)
+
+export const getRemovedListItemDetails = (
+  instanceChange: ModificationChange<InstanceElement>,
+): {
+  removedListItems: string[]
+  elemID: ElemID
+  detailedMessage: string
+} => {
+  const getters = getGettersByType(instanceChange.data.before.elemID.typeName)
+  if (getters === undefined) {
+    return { removedListItems: [], elemID: instanceChange.data.before.elemID, detailedMessage: '' }
+  }
+  const { before, after } = instanceChange.data
+  const beforeItemList = getIdentifierList(before, getters)
+  const afterItemSet = new Set<string>(getIdentifierList(after, getters))
+  const removedListItems = beforeItemList.filter(id => !afterItemSet.has(id))
+  return {
+    removedListItems,
+    elemID: before.elemID,
+    detailedMessage: getters.getDetailedMessage(removedListItems),
+  }
+}
+
+const changeValidator: NetsuiteChangeValidator = async changes => {
+  const instanceChanges = await awu(changes)
+    .filter(isModificationChange)
+    .filter(isInstanceChange)
+    .toArray() as ModificationChange<InstanceElement>[]
+
+  return instanceChanges
+    .map(getRemovedListItemDetails)
+    .filter(({ removedListItems }) => !_.isEmpty(removedListItems))
+    .map(({ detailedMessage, elemID }) => ({
```

Comment:
we should try not using `as` (line 167) - this should replace it
```suggestion
    .map(({ detailedMessage, elemID }): ChangeError => ({
```